src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1625,97 +1649,235 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
-            }
-
-            else if (!string.IsNullOrEmpty(request.IdToken)) {
-                ticket = await ReceiveIdentityTokenAsync(request.IdToken, request);
-            }
-
-            else if (!string.IsNullOrEmpty(request.GetRefreshToken())) {
-                ticket = await ReceiveRefreshTokenAsync(request.GetRefreshToken(), request);
-            }
 
-            else {
-                await SendErrorPayloadAsync(new OpenIdConnectMessage {
+            if (string.IsNullOrWhiteSpace(request.Token))
+            {
+                await SendErrorPayloadAsync(new OpenIdConnectMessage
+                {
                     Error = OpenIdConnectConstants.Errors.InvalidRequest,
-                    ErrorDescription = "A malformed validation request has been received: " +
-                        "either an identity token, an access token or a refresh token must be provided."
+                    ErrorDescription = "A malformed introspection request has been received: " +
+                        "a 'token' parameter with an access, refresh, or id token is required."
                 });
 
                 return;
             }
 
-            if (ticket == null) {
-                Options.Logger.WriteError("invalid token");
+            //TODO: Re-examine authentication logic to ensure authentication happens in some way.
+            //TODO: Allow extensibility to skip authentication somehow for access tokens.
+            //      Perhaps in Options.Provider.ValidationEndpoint()?
+            // Attempt to Recieve an Access Token for authentication
+            // See https://tools.ietf.org/html/rfc7662#section-2.1
+            var authTicket = await ReceiveAccessTokenAsync(request.Token, request);
+
+            // If the access token could not be recieved or the ticket could not be parsed,
+            // Attempt to validate based on client credentials
+            if(authTicket == null){
+                var clientNotification = new ValidateClientAuthenticationContext(Context, Options, request);
+                await Options.Provider.ValidateClientAuthentication(clientNotification);
+    
+                // Reject the request if client authentication was rejected.
+                if (!clientNotification.IsValidated)
+                {
+                    Options.Logger.WriteError("invalid client authentication.");
+    
+                    await SendErrorPayloadAsync(new OpenIdConnectMessage
+                    {
+                        Error = clientNotification.Error ?? OpenIdConnectConstants.Errors.InvalidClient,
+                        ErrorDescription = clientNotification.ErrorDescription,
+                        ErrorUri = clientNotification.ErrorUri
+                    });
+    
+                    return;
+                }
+    
+                // Reject if client authentication was skipped.
+                if (clientNotification.IsSkipped && request.IsClientCredentialsGrantType())
```

Comment:
The specs are clearly too strict here and assume the token are short enough to be prone to scanning attacks. In OSOS/ASOS, access tokens use JWT by default and the refresh tokens are protected by the data protection block. Both methods result in loooong tokens, which makes this kind of attacks very unlikely :smile: 

The new `Skipped()` support should be enough.